src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -884,23 +884,36 @@ void IslandWindow::_RestoreFullscreenPosition(const RECT rcWork)
                rcWork.left - _rcWorkBeforeFullscreen.left,
                rcWork.top - _rcWorkBeforeFullscreen.top);
 
+    const til::size ncSize{ GetTotalNonClientExclusiveSize(dpiWindow) };
+
+    RECT rcWorkAdjusted = rcWork;
+
+    // GH#10199 - adjust the size of the "work" rect by the size of our borders.
+    // We want to make sure the window is restored within the bounds of the
+    // monitor we're on, but it's totally fine if the invisible borders are
+    // outside the monitor.
+    rcWorkAdjusted.left -= ncSize.width<long>() / 2;
+    rcWorkAdjusted.right += ncSize.width<long>() / 2;

```

Comment:
Could width or height reasonably be an odd number where this won't work properly? Or will the offsetting below correct that to slide the whole thing 1px into place at the end then?

I think this is fine... but this is where my brain is tying itself into a knot.

Reply:
Form my experience, these numbers are always (8 * dpiScale), so I don't think they can be. IIRC the borders from DWM (the ones that are causing the problem here) are always "8px"

Reply:
Oh well pffsh. OK.